File: airbyte-server/src/test/java/io/airbyte/server/apis/DeploymentMetadataApiControllerTest.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.server.apis;

import static org.mockito.Mockito.when;

import io.airbyte.api.model.generated.DeploymentMetadataRead;
import io.airbyte.config.Configs;
import io.micronaut.context.annotation.Requires;
import io.micronaut.context.env.Environment;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.HttpStatus;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import java.util.UUID;
import org.junit.jupiter.api.Test;

@MicronautTest
@Requires(env = {Environment.TEST})
@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
class DeploymentMetadataApiControllerTest extends BaseControllerTest {

  @Test
  void testFetchDeploymentMetadata() {
    final DeploymentMetadataRead deploymentMetadataRead = new DeploymentMetadataRead()
        .id(UUID.randomUUID())
        .environment(Configs.WorkerEnvironment.KUBERNETES.name())
        .mode(Configs.DeploymentMode.OSS.name())
        .version("0.2.3");
    when(deploymentMetadataHandler.getDeploymentMetadata()).thenReturn(deploymentMetadataRead);
    final String path = "/api/v1/deployment/metadata";
    testEndpointStatus(
        HttpRequest.POST(path, null),
        HttpStatus.OK);
  }

}

File: airbyte-server/src/test/kotlin/io/airbyte/server/apis/controllers/DeploymentMetadataApiControllerTest.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.server.apis.controllers

import io.airbyte.api.model.generated.DeploymentMetadataRead
import io.airbyte.commons.server.handlers.DeploymentMetadataHandler
import io.mockk.every
import io.mockk.mockk
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.util.UUID

class DeploymentMetadataApiControllerTest {
  private lateinit var deploymentMetadataHandler: DeploymentMetadataHandler
  private lateinit var controller: DeploymentMetadataApiController

  @BeforeEach
  fun setup() {
    deploymentMetadataHandler = mockk()
    controller = DeploymentMetadataApiController(deploymentMetadataHandler)
  }

  @Test
  fun testFetchDeploymentMetadata() {
    val expected =
      DeploymentMetadataRead()
        .id(UUID.randomUUID())
        .mode("COMMUNITY")
        .version("0.2.3")

    every { deploymentMetadataHandler.getDeploymentMetadata() } returns expected

    val result = controller.getDeploymentMetadata()
    assertEquals(expected, result)
  }
}
